src/structures/InviteGuild.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const BaseGuild = require('./BaseGuild');
+const WelcomeScreen = require('./WelcomeScreen');
+
+/**
+ * Represents a guild received from an invite, including welcome screen data if available.
+ * @extends {BaseGuild}
+ */
+class InviteGuild extends BaseGuild {
```

Comment:
The following properties are available to `InviteGuild` but not to `BaseGuild` (due to them not being available to `OAuth2Guild`):
`splash`, `banner`, `description`, `verification_level`, `vanity_url_code`, `nsfw_level`

I don't know if another abstraction layer is worth the code deduplication (and I wouldn't know what to name that middle-guild-class).

Reply:
After more internal discussion we went with the intermediary guild, called AnonymousGuild for now, if other names come to mind these are easily swappable.

Reply:
> 
> 
> Currently when fetching and invite the following error is being thrown:
> 
> ```js
> TypeError: Cannot read property 'cache' of undefined
>     at Function.create (discord.js\src\structures\Channel.js:156:37)
>     at ChannelManager.add (\discord.js\src\managers\ChannelManager.js:30:29)
>     at Invite._patch (\discord.js\src\structures\Invite.js:110:41)
>     at new Invite (\discord.js\src\structures\Invite.js:16:10)
>     at \discord.js\src\client\Client.js:259:21
> ```
> 
> The line in question is:
> 
> https://github.com/discordjs/discord.js/blob/65dc00f3210065015684b6d585f6747bd5ebadf1/src/structures/Channel.js#L156
> 
> This could potentially be solved by checking whether the guild has a channel manager (and cache) e.g. `&& guild.channels?.cache` or maybe through an instanceof check. since only `Guild`s have a channel cache. (No idea about circulars here though)

This has been addressed (as well for guilds the bot is in) through optional chaining in a83c0db and should be fixed (could at least no longer reproduce this issue)